### DISCUSSION TITLE
How do I code a yield constraint?

### DISCUSSION THREAD

**Discussion by 0blue00** [OptimizationExpert/Pyomo#7] [Q&A] [answered]
I have n items.
I have an array item_val of size n with item values for the n items.
My decision variable (item_picked) is a binary array of size n, specifying if I have picked the item or not.
I want to code a constraint where I find the minimum item value of the items picked; and for each item picked, the minimum divided by the item value should be greater than 0.7

Example:
n = 5
item_val = [1, 2, 3, 5, 2]
item_picked = [1, 1, 0, 0, 1]

here,
minimum item value among items picked is: min_val = min(item_val[0], item_val[1], item_val[4]) = min(1,2,2) = 1
constraint should be:
min_val/item_val[0] > 0.7
min_val/item_val[1] > 0.7
min_val/item_val[4] > 0.7

Please help me code this constraint in pyomo. 
TIA

**Comment by OptimizationExpert** [comment, marked as answer]
https://github.com/OptimizationExpert/Pyomo/blob/main/yield.ipynb
